docs/refactoring/confirmation-pages-routing/README.md

```diff
@@ -0,0 +1,65 @@
+# Refactoring - Confirmation pages routing
+
+This document details how routing to confirmation pages is currently done and the proposed improvements in routing.
+
+## Current flow
+
+The current flow of routing to confirmation pages is un-necessarily complicated and have issues.
+
+![Confirmation Pages Routing - Current](https://raw.githubusercontent.com/MetaMask/metamask-extension/develop/docs/refactoring/confirmation-pages-routing/current.png)
+
+- There are 2 ways in which confirmation pages can be opened:
+  1. User triggers send flow from within Metamask
+     - If the user triggers the send flow from within MetaMask and selects the recipient and amount on the send screen, an unapproved transaction is created in the background and the user is redirected to the **/confirm-transaction** route.
+  2. DAPP sends request to Metamask
+     - If DAPP sends request to Metamask an un-approved transaction or signature request is created in background and UI is triggered open (if it is not already open).
```

Comment:
```suggestion
     - If DAPP sends request to Metamask an unapproved transaction or signature request is created in background and UI is triggered open (if it is not already open).
```

Reply:
I fixed it in this following PR: https://github.com/MetaMask/metamask-extension/pull/17821